lib/advanced_computer_receiver.rb

```diff
@@ -0,0 +1,111 @@
+class AdvancedComputerReceiver
+  def initialize(board, game_checker)
+    @board = board
+    @game_checker = game_checker
+  end
+
+  def get_player_input(validator, display, computer_player, human_player)
+    calculate_best_move(validator, computer_player, human_player)
+  end
+
+  private
+
+  def calculate_best_move(validator, computer_player, human_player)
+    # need to determine best move.
+    # computer win
+    # stop human win
+    # fork
+    # block fork
+    # center
+    # opposite corner
+    # empty corner
+    # empty side
+    corners = [1, 3, 7, 9]
+    sides = [2, 4, 6, 8]
+
+    if check_function(validator, computer_player, :winner?)
+      #   puts "computer win"
+      check_function(validator, computer_player, :winner?)
```

Comment:
I'm sure you had the puts statements for testing but good to remove when you PR

Reply:
Why do you call check_function twice - as part of if statement and within if statement?